src/main/java/lotto/domain/LottoAnnounce.java

```diff
@@ -0,0 +1,29 @@
+package lotto.domain;
+
+import java.util.List;
+import java.util.stream.Collectors;
+import lotto.domain.constant.LottoRank;
+
+public class LottoAnnounce {
+
+    private final Lotto winningNumber;
+
+    public LottoAnnounce(Lotto winningNumber) {
+        this.winningNumber = winningNumber;
+    }
+
+    public LottoRanks createRanks(List<Lotto> lottos) {
+        return new LottoRanks(lottos.stream()
+            .map(this::scoreHit)
+            .map(LottoRank::toWinLotto)
+            .collect(Collectors.toList()));
+    }
+
+    private int scoreHit(Lotto lotto) {
+        return (int) lotto.getNumbers()
+            .stream()
+            .filter(number -> winningNumber.getNumbers().contains(number))
```

Comment:
getter로 값을 꺼내는 대신에, 메시지를 보내서 처리해보면 어떨까요?
```suggestion
            .filter(number -> winningNumber.contains(number))
```